src/pip/_internal/utils/misc.py

```diff
@@ -420,17 +420,21 @@ def renames(old, new):
             pass
 
 
-def is_local(path):
-    # type: (str) -> bool
+def is_local(path, normalized=False):
```

Comment:
Rather than complicating this function with a flag, and since I only count this function being called twice in the non-test code base, I would just document in the `:param:` that the path passed to the function must have had `normalize_path()` called on it, and then make sure that `normalize_path()` has been applied at each call site.

Also, I see that one of the two calls in the code base is directly passing the return value of `dist_location()`. Except for that one spot, it looks like every other call to `dist_location()` is calling `normalize_path()` on the return value of `dist_location()` anyways. So maybe either as part of this PR or as a separate PR, `dist_location()` can be changed to return a path that has had `normalize_path()` called on it. And that would make the code strictly simpler.